pkgs/development/libraries/gjs/default.nix

```diff
@@ -107,12 +107,12 @@ in stdenv.mkDerivation rec {
 
   postInstall = ''
     # TODO: make the glib setup hook handle moving the schemas in other outputs.
-    installedTestsSchemaDatadir="$installedTests/share/gsettings-schemas/${pname}-${version}"
+    installedTestsSchemaDatadir="$installedTests/share/gsettings-schemas/gjs-${finalAttrs.version}"
     mkdir -p "$installedTestsSchemaDatadir"
     mv "$installedTests/share/glib-2.0" "$installedTestsSchemaDatadir"
   '';
 
-  postFixup = ''
+  postFixup = lib.optionalString finalAttrs.finalPackage.doCheck ''
```

Comment:
Installed tests are orthogonal to `doCheck`.

Reply:
I think this is safer instead of duplicating the condition from doCheck in case someone does an overwrite which uses doCheck = false like the nox overlay.

Reply:
Well, reusing the unrelated variable makes the code more confusing.

Reply:
How should I then build the override in the module? I can't just overwrite postFix up to empty string or null.

Reply:
Maybe add `reduceClosure` argument or something?

Reply:
That would be even more confusing and not very self descriptive.

Or did I misinterpret that those deps are only used for the tests in this package?

Reply:
They are. But installed tests are a separate build output that will be executed by host and need a different way to declare dependencies than build tests.

Reply:
I've added a new option to do that